### PR TITLE
Don't fail TestLauncher execution when only tasks are configured

### PR DESCRIPTION
**Pull request by donat**

### CODE REVIEW

File: subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestExecutionResultEvaluator.java [hunk 1]
```diff
@@ -64,7 +64,21 @@ public TestExecutionResultEvaluator(
...
-        return resultCount.get() == 0;
+        if (noTestsSelected()) {
+             return false;
+        } else {
+            return resultCount.get() == 0;
+        }
+    }
+
+    private boolean noTestsSelected() {
+        return internalTestExecutionRequest.getTestExecutionDescriptors().isEmpty()
```
Comment (hegyibalint) [inline subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestExecutionResultEvaluator.java:75]: 🤔 Maybe this should be broken down to named `boolean` values, and combining them at the end?
This is not that understandable nor readable.

**Review by hegyibalint — APPROVED** [review]

File: subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestExecutionResultEvaluator.java (hunk 1 above)
Comment (donat) [reply subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestExecutionResultEvaluator.java]: I've refactored this; PTAL

**Comment by donat** [comment]
@bot-gradle test RFN

**Comment by donat** [comment]
@bot-gradle test RFN

**Comment by donat** [comment]
@bot-gradle test and merge

**Pull request merged**